packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  deleteProject as deleteGen1Project,
+  deleteProjectDir,
+  initJSProjectWithProfile,
+  addAuthWithDefault,
+  amplifyPushAuth,
+  getProjectMeta,
+  getUserPool,
+} from '@aws-amplify/amplify-e2e-core';
+import { npmInstall, getNpxPath, nspawn as spawn } from '@aws-amplify/amplify-e2e-core';
+import * as fs from 'fs-extra';
+import { $TSAny } from '@aws-amplify/amplify-cli-core';
+import path from 'node:path';
+import assert from 'node:assert';
+import { CloudControlClient, GetResourceCommand } from '@aws-sdk/client-cloudcontrol';
+
+const pushTimeoutMS = 1000 * 60 * 20; // 20 minutes;
+
+async function getResourceDetails(typeName: string, identifier: string, region: string) {
+  const client = new CloudControlClient({ region });
+  const command = new GetResourceCommand({
+    TypeName: typeName,
+    Identifier: identifier,
+  });
+
+  try {
+    const response = await client.send(command);
+    return JSON.parse(response.ResourceDescription.Properties);
+  } catch (error) {
+    console.error('Error fetching resource details:', error);
+    throw error;
+  }
+}
+
+function runGen2SandboxCommand(cwd: string) {
+  try {
+    npmInstall(cwd);
+  } catch (error) {
+    console.error('Failed to install dependencies:', error);
+    throw error;
+  }
+  return spawn(getNpxPath(), ['ampx', 'sandbox', '--once'], {
+    cwd,
+    stripColors: true,
+    noOutputTimeout: pushTimeoutMS,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+  }).runAsync();
+}
+
+function runCodegenCommand(cwd: string) {
+  return spawn(getNpxPath(), ['@aws-amplify/migrate', 'to-gen-2', 'generate-code'], {
+    cwd,
+    stripColors: true,
+    noOutputTimeout: pushTimeoutMS,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+  }).runAsync();
+}
+
+function deleteGen2Sandbox(cwd: string) {
+  return spawn(getNpxPath(), ['ampx', 'sandbox', 'delete'], {
+    cwd,
+    stripColors: true,
+    noOutputTimeout: pushTimeoutMS,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+  })
+    .wait("Are you sure you want to delete all the resources in your sandbox environment (This can't be undone)?")
+    .sendConfirmYes()
+    .wait('Finished deleting.')
+    .runAsync();
+}
+
+export async function setupGen1Project(projRoot: string, projectName: string) {
```

Comment:
```suggestion
export async function setupAndPushGen1Project(projRoot: string, projectName: string) {
```

push is heavy long running operation. please be explicit in function name.